src/PrebidMobile/PrebidMobile/AdUnits/Utils.swift

```diff
@@ -89,6 +90,157 @@ public class Utils: NSObject {
 
     }
 }
+    
+    public func findPrebidCreativeSize(_ adView: UIView, completion: @escaping (CGSize?) -> Void) {
+        
+        let view = self.recursivelyFindWebView(adView) { (subView) -> Bool in
+            return subView is WKWebView || subView is UIWebView
+        }
+        
+        if let wkWebView = view as? WKWebView  {
+            Log.debug("subView is WKWebView")
+            self.findSizeInWebViewAsync(wkWebView: wkWebView, completion: completion)
+            
+        } else if let uiWebView = view as? UIWebView {
+            Log.debug("subView is UIWebView")
+            self.findSizeInWebViewAsync(uiWebView: uiWebView, completion: completion)
+        } else {
+            Log.warn("subView doesn't include WebView")
+        }
+       
+    }
+    
+    func runResizeCompletion(size: CGSize?, completion: @escaping (CGSize?) -> Void) {
+        
+        Log.debug("size:\(size)")
+        completion(size)
+    }
+    
+    func recursivelyFindWebView(_ view: UIView, closure:(UIView) -> Bool) -> UIView? {
+        for subview in view.subviews {
+            
+            if closure(subview)  {
+                return subview
+            }
+            
+            if let result = recursivelyFindWebView(subview, closure: closure) {
+                return result
+            }
+        }
+        
+        return nil
+    }
+    
+    func findSizeInWebViewAsync(wkWebView: WKWebView, completion: @escaping (CGSize?) -> Void) {
+        
+        wkWebView.evaluateJavaScript("document.body.innerHTML", completionHandler: { (value: Any!, error: Error!) -> Void in
+            
+            if error != nil {
+                Log.warn("error:\(error.localizedDescription)")
+                return
+            }
+
+            let wkResult = self.findSizeInJavaScript(jsCode: value as? String)
+            
+            self.runResizeCompletion(size: wkResult, completion: completion)
+        })
+        
+    }
+    
+    func findSizeInWebViewAsync(uiWebView: UIWebView, completion: @escaping (CGSize?) -> Void) {
+        
+        Timer.scheduledTimer(withTimeInterval: 0.1, repeats: true) { (timer) in
```

Comment:
This API only works for iOS 10 & above and not for iOS 9. Since we are downgrading to iOS 9, can you fix this?

Reply:
Yes, I will upgrade the PR.